operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -392,20 +392,250 @@ Once this step has completed:
+        ```
+
+    1. Verify that the storage node booted and is configured correctly. The CFS configuration can be checked will the command below using the `xname` of the node that was upgraded.
+
+        ```bash
+        XNAME=x3000c0s13b0n0
+        cray cfs components describe $XNAME
+        ```
+
+        It is expected that `configurationStatus` is `configured`. If it is `pending`, then wait for the status to be `configured`.
+
+    1. (`ncn-m#`) Upgrade the remaining storage nodes. This will upgrade them serially.
+
+        **NOTE** This creates an additional, separate Argo workflow for upgrading NCN storage nodes. The Argo workflow name will include the string `ncn-lifecycle-rebuild`. If monitoring progress with the Argo UI, remember to include these workflows.
+
+        ```bash
+        /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ncn-s002,ncn-s003,ncn-s004 --upgrade
+        ```
+
+1. NCN master node upgrade (`ncn-m002` and `ncn-m003`)
+
+    1. Use `kubectl` to label `ncn-m003` node with `iuf-prevent-rollout=true` to ensure `management-nodes-rollout` only rebuilds a single NCN master node, `ncn-m002`.
+
+        (`ncn-m001#`) Label `ncn-m003` to prevent it from rebuilding.
+
+        ```bash
+        kubectl label nodes "ncn-m003" --overwrite iuf-prevent-rollout=true
+        ```
+
+        (`ncn-m001#`) Verify the IUF node labels are present on the desired node.
+
+        ```bash
+        kubectl get nodes --show-labels | grep iuf-prevent-rollout
+        ```
+
+    1. Invoke `iuf run` with `-r` to execute the [`management-nodes-rollout`](../stages/management_nodes_rollout.md) stage on `ncn-m002`. This will rebuild the canary node with the new CFS configuration and image built in
+    previous steps of the workflow.
+
+        (`ncn-m001#`) Execute the `management-nodes-rollout` stage with `ncn-m002`.
+
+        ```bash
+        iuf -a "${ACTIVITY_NAME}" run -r management-nodes-rollout --limit-management-rollout Management_Master
+        ```
+
+    1. Verify that `ncn-m002` booted successfully with the desired image and CFS configuration.
+
+    1. Use `kubectl` to label remove `iuf-prevent-rollout=true` from `ncn-m003` and add it to `ncn-m002`.
+
+        (`ncn-m001#`) Remove label from `ncn-m003` and add it to `ncn-m002` to prevent it from rebuilding.
+
+        ```bash
+        kubectl label nodes "ncn-m002" --overwrite iuf-prevent-rollout=true
+        kubectl label nodes "ncn-m003" --overwrite iuf-prevent-rollout-
+        ```
+
+        (`ncn-m001#`) Verify the IUF node labels are present on the desired node.
+
+        ```bash
+        kubectl get nodes --show-labels | grep iuf-prevent-rollout
+        ```
+
+    1. Invoke `iuf run` with `-r` to execute the [`management-nodes-rollout`](../stages/management_nodes_rollout.md) stage on `ncn-m003`. This will rebuild the `ncn-m002` with the new CFS configuration and image built in
```

Comment:
```suggestion
    1. Invoke `iuf run` with `-r` to execute the [`management-nodes-rollout`](../stages/management_nodes_rollout.md) stage on `ncn-m003`. This will rebuild the `ncn-m003` with the new CFS configuration and image built in
```